src/combat.cpp

```diff
@@ -1086,9 +1088,9 @@ void ValueCallback::getMinMaxValues(Player* player, CombatDamage& damage) const
 
 		case COMBAT_FORMULA_SKILL: {
 			// onGetPlayerMinMaxValues(player, attackSkill, attackValue, attackFactor)
-			Item* tool = player->getWeapon();
-			const Weapon* weapon = g_weapons->getWeapon(tool);
-			Item* item = nullptr;
+			auto tool = player->getWeapon();
+			auto weapon = g_weapons->getWeapon(tool);
+			std::shared_ptr<Item> item = nullptr;
```

Comment:
Could this be const auto
```suggestion
			const auto tool = player->getWeapon();
			const auto weapon = g_weapons->getWeapon(tool);
			auto item = std::shared_ptr<Item>(nullptr);
```